src/main/java/org/edumips64/utils/cli/RunCommand.java

```diff
@@ -0,0 +1,44 @@
+package org.edumips64.utils.cli;
+
+import org.edumips64.core.is.HaltException;
+import org.edumips64.utils.CurrentLocale;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.ParentCommand;
+
+@Command(name = "run", resourceBundle = "CliMessages")
+public class RunCommand implements Runnable {
+
+    @ParentCommand
+    private Cli cli;
+
+    @Override
+    public void run() {
+        if (cli.isCpuRunnable()) {
+            execute();
+        } else {
+            cli.printNotRunnable();
+        }
+    }
+
+    private void execute() {
+        int steps = 0;
+        long startTimeMs = System.currentTimeMillis();
+        try {
+            while(true) {
+                steps++;
+                cli.getCPU().step();
+            }
+        } catch (HaltException e) {
+            long endTimeMs = System.currentTimeMillis();
+            long totalTimeMs = endTimeMs - startTimeMs;
+            printEndOfExecMsg(steps, totalTimeMs);
```

Comment:
Nice to give some feedback about the execution time! :)

Reply:
Thanks!